Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -617,65 +639,75 @@ void SourceBufferPrivate::didReceiveSample(Ref<MediaSample>&& sample)
     if (!isAttached())
         return;
 
-    if (m_pendingAppendOperations.isEmpty() || !std::holds_alternative<SamplesVector>(m_pendingAppendOperations.last()))
-        m_pendingAppendOperations.append({ SamplesVector { } }); // This is a new operation.
+    if (m_pendingOperations.isEmpty() || !std::holds_alternative<SamplesVector>(m_pendingOperations.last()))
+        m_pendingOperations.append({ SamplesVector { } }); // This is a new operation.
 
     DEBUG_LOG(LOGIDENTIFIER, sample.get());
-    std::get<SamplesVector>(m_pendingAppendOperations.last()).append(WTFMove(sample));
+    std::get<SamplesVector>(m_pendingOperations.last()).append(WTFMove(sample));
 }
 
 void SourceBufferPrivate::append(Ref<SharedBuffer>&& buffer)
 {
-    append(buffer->extractData());
+    queueOperation(WTFMove(buffer));
 }
 
-void SourceBufferPrivate::append(Vector<unsigned char>&&)
+void SourceBufferPrivate::queueOperation(Operation&& operation)
 {
-    RELEASE_ASSERT_NOT_REACHED();
+    m_pendingOperations.append(WTFMove(operation));
+    processPendingOperations();
 }
 
 void SourceBufferPrivate::processPendingOperations()
 {
-    if (m_pendingInitOperation)
-        return;
-
-    while (!m_pendingAppendOperations.isEmpty()) {
-        if (!isAttached())
-            break;
-
-        auto operation = m_pendingAppendOperations.takeFirst();
+    while (!m_pendingOperations.isEmpty()) {
+        if (!isAttached() || m_errored) {
+            abortPendingOperations();
+            return;
+        }
+        if (m_didReceiveInitializationSegmentErrored || m_didReceiveSampleErrored)
+            m_pendingOperations.prepend(ErrorOperation { });
+        else if (m_operationState != OperationState::Idle)
+            return;
+        auto operation = m_pendingOperations.takeFirst();
         std::visit(WTF::makeVisitor([&](InitOperation&& initOperation) {
-            processInitOperation(WTFMove(initOperation));
+            processOperation(WTFMove(initOperation));
```

Comment:
I tend to prefer more specific names like processInitOperation, it helps for searching

Reply:
re-renamed the methods